modules/uhttp/client/client_filters.go

```diff
@@ -86,6 +86,10 @@ func authenticationFilter(
 	ctx fx.Context, req *http.Request, next BasicClient,
 ) (resp *http.Response, err error) {
 	authClient := auth.Instance()
+
+	// Client needs to know what service it is to authenticate
+	authClient.SetAttribute(ctx, auth.ServiceAuth, _serviceName)
```

Comment:
Yes, we need to set it every time on the context in order for authenticate to work. (ctx is new per request)